src/Integration/Laravel/Queue/src/AsyncAwsSqsQueue.php

```diff
@@ -0,0 +1,167 @@
+     *
+     * @var string
+     */
+    private $suffix;
+
+    /**
+     * Create a new Amazon SQS queue instance.
+     *
+     * @param  \AsyncAws\Sqs\SqsClient  $sqs
+     * @param  string  $default
+     * @param  string  $prefix
+     * @param  string  $suffix
+     * @return void
+     */
+    public function __construct(SqsClient $sqs, $default, $prefix = '', $suffix = '')
+    {
+        $this->sqs = $sqs;
+        $this->prefix = $prefix;
+        $this->default = $default;
+        $this->suffix = $suffix;
+    }
+    /**
+     * Get the size of the queue.
+     *
+     * @param  string|null  $queue
+     * @return int
+     */
+    public function size($queue = null)
+    {
+        $response = $this->sqs->getQueueAttributes([
+            'QueueUrl' => $this->getQueue($queue),
+            'AttributeNames' => [QueueAttributeName::APPROXIMATE_NUMBER_OF_MESSAGES],
+        ]);
+
+        $attributes = $response->getAttributes();
+
+        return (int) $attributes['ApproximateNumberOfMessages'];
+    }
+
+    /**
+     * Push a new job onto the queue.
+     *
+     * @param  string  $job
+     * @param  mixed  $data
+     * @param  string|null  $queue
+     * @return mixed
+     */
+    public function push($job, $data = '', $queue = null)
+    {
+        return $this->pushRaw($this->createPayload($job, $queue ?: $this->default, $data), $queue);
+    }
+
+    /**
+     * Push a raw payload onto the queue.
+     *
+     * @param  string  $payload
+     * @param  string|null  $queue
+     * @param  array  $options
+     * @return mixed
+     */
+    public function pushRaw($payload, $queue = null, array $options = [])
+    {
+        return $this->sqs->sendMessage([
+            'QueueUrl' => $this->getQueue($queue), 'MessageBody' => $payload,
+        ])->getMessageId();
+    }
+
+    /**
+     * Push a new job onto the queue after a delay.
+     *
+     * @param  \DateTimeInterface|\DateInterval|int  $delay
+     * @param  string  $job
+     * @param  mixed  $data
+     * @param  string|null  $queue
+     * @return mixed
+     */
+    public function later($delay, $job, $data = '', $queue = null)
+    {
+        return $this->sqs->sendMessage([
+            'QueueUrl' => $this->getQueue($queue),
+            'MessageBody' => $this->createPayload($job, $queue ?: $this->default, $data),
+            'DelaySeconds' => $this->secondsUntil($delay),
+        ])->getMessageId();
+    }
+
+    /**
+     * Pop the next job off of the queue.
+     *
+     * @param  string|null  $queue
+     * @return \Illuminate\Contracts\Queue\Job|null
+     */
+    public function pop($queue = null)
+    {
+        $response = $this->sqs->receiveMessage([
+            'QueueUrl' => $queue = $this->getQueue($queue),
+            'AttributeNames' => ['ApproximateReceiveCount'],
+        ]);
+
+        foreach($response->getMessages() as $message) {
+            return new AsyncAwsSqsJob(
+                $this->container, $this->sqs, $message,
```

Comment:
where does `$this->container` come from?

Reply:
`Illuminate\Queue\Queue::setContainer()`